packages/next/src/next-devtools/dev-overlay/components/overview/segment-boundary-trigger.tsx

```diff
@@ -102,63 +28,69 @@ export function SegmentBoundaryTrigger({
     icon: <ResetIcon />,
   }
 
-  const handleSelect = (value: string) => {
-    if (value === 'not-found') {
-      onSelectBoundary('not-found')
-    } else if (value === 'loading') {
-      onSelectBoundary('loading')
-    } else if (value === 'error') {
-      onSelectBoundary('error')
-    } else if (value === 'reset') {
-      onSelectBoundary(null)
-    }
-    setIsOpen(false)
-  }
+  const handleSelect = useCallback(
+    (value: string) => {
+      if (value === 'not-found') {
+        onSelectBoundary('not-found')
+      } else if (value === 'loading') {
+        onSelectBoundary('loading')
+      } else if (value === 'error') {
+        onSelectBoundary('error')
+      } else if (value === 'reset') {
+        onSelectBoundary(null)
+      }
+    },
+    [onSelectBoundary]
+  )
```

Comment:
this feels a bit easier clearer to me but not a blocker
```suggestion
const handleSelect = useCallback(
  (value: string) => {
    switch (value) {
      case 'not-found':
      case 'loading':
      case 'error':
        onSelectBoundary(value)
        break
      case 'reset':
        onSelectBoundary(null)
        break
    }
  },
  [onSelectBoundary]
)
```